lib/puppet/metatype/manager.rb

```diff
@@ -146,16 +156,19 @@ def type(name)
     # We are overwhelmingly symbols here, which usually match, so it is worth
     # having this special-case to return quickly.  Like, 25K symbols vs. 300
     # strings in this method. --daniel 2012-07-17
-    return @types[name] if @types[name]
+    return @types[name] if @types.include? name
```

Comment:
Isn't `@types` here a Hash? If so, `include?` is a hash lookup, which is hard to beat with a faster construct. One can experiment if one of the methods like `key?(name)` is faster (it is probably the very same thing though).